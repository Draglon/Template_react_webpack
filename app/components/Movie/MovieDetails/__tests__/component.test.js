import React from 'react';

import MovieDetailsComponent from '../component';

describe('<MovieDetailsComponent />', () => {
  const props = {
    title: 'some title',
    movieDetails: [],
  };
  const component = shallow(<MovieDetailsComponent {...props} />);

  describe('should match snapshots', () => {
    it('should match snapshot - without movieDetails', () => {
      expect(component).toMatchSnapshot();
    });

    it('should match snapshot - movieDetails (character)', () => {
      const newProps = {
        ...props,
        movieDetails: [
          { id: 1, profile_path: 'path_1', name: 'name_1', character: 'character_1' },
          { id: 2, profile_path: 'path_2', name: 'name_2', character: 'character_2' },
        ],
      };
      component.setProps(newProps);
      expect(component).toMatchSnapshot();
    });

    it('should match snapshot - movieDetails (job)', () => {
      const newProps = {
        ...props,
        movieDetails: [
          { id: 1, profile_path: 'path_1', name: 'name_1', job: 'job_1' },
          { id: 2, profile_path: 'path_2', name: 'name_2', job: 'job_2' },
        ],
      };
      component.setProps(newProps);
      expect(component).toMatchSnapshot();
    });
  });
});
