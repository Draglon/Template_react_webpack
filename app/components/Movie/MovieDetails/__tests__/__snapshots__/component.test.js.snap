// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieDetailsComponent /> should match snapshots should match snapshot - movieDetails (character) 1`] = `
<Fragment>
  <Row
    gutter={0}
  >
    <Col
      className="top-margin"
      offset={2}
      span={10}
    >
      <Title
        level={3}
      >
        some title
      </Title>
    </Col>
  </Row>
  <Row
    gutter={8}
    type="flex"
  >
    <Col
      className="movie__details"
      offset={2}
      span={20}
    >
      <Col
        key="1"
        lg={
          Object {
            "span": 4,
          }
        }
        md={
          Object {
            "span": 6,
          }
        }
        sm={
          Object {
            "span": 8,
          }
        }
        xl={
          Object {
            "span": 4,
          }
        }
        xs={
          Object {
            "span": 12,
          }
        }
      >
        <Card
          className="top-margin"
          cover={
            <ImageComponent
              alt="name_1"
              className="movie-image"
              path="path_1"
            />
          }
        >
          <Meta
            description="character_1"
            title="name_1"
          />
        </Card>
      </Col>
      <Col
        key="2"
        lg={
          Object {
            "span": 4,
          }
        }
        md={
          Object {
            "span": 6,
          }
        }
        sm={
          Object {
            "span": 8,
          }
        }
        xl={
          Object {
            "span": 4,
          }
        }
        xs={
          Object {
            "span": 12,
          }
        }
      >
        <Card
          className="top-margin"
          cover={
            <ImageComponent
              alt="name_2"
              className="movie-image"
              path="path_2"
            />
          }
        >
          <Meta
            description="character_2"
            title="name_2"
          />
        </Card>
      </Col>
    </Col>
  </Row>
</Fragment>
`;

exports[`<MovieDetailsComponent /> should match snapshots should match snapshot - movieDetails (job) 1`] = `
<Fragment>
  <Row
    gutter={0}
  >
    <Col
      className="top-margin"
      offset={2}
      span={10}
    >
      <Title
        level={3}
      >
        some title
      </Title>
    </Col>
  </Row>
  <Row
    gutter={8}
    type="flex"
  >
    <Col
      className="movie__details"
      offset={2}
      span={20}
    >
      <Col
        key="1"
        lg={
          Object {
            "span": 4,
          }
        }
        md={
          Object {
            "span": 6,
          }
        }
        sm={
          Object {
            "span": 8,
          }
        }
        xl={
          Object {
            "span": 4,
          }
        }
        xs={
          Object {
            "span": 12,
          }
        }
      >
        <Card
          className="top-margin"
          cover={
            <ImageComponent
              alt="name_1"
              className="movie-image"
              path="path_1"
            />
          }
        >
          <Meta
            description="job_1"
            title="name_1"
          />
        </Card>
      </Col>
      <Col
        key="2"
        lg={
          Object {
            "span": 4,
          }
        }
        md={
          Object {
            "span": 6,
          }
        }
        sm={
          Object {
            "span": 8,
          }
        }
        xl={
          Object {
            "span": 4,
          }
        }
        xs={
          Object {
            "span": 12,
          }
        }
      >
        <Card
          className="top-margin"
          cover={
            <ImageComponent
              alt="name_2"
              className="movie-image"
              path="path_2"
            />
          }
        >
          <Meta
            description="job_2"
            title="name_2"
          />
        </Card>
      </Col>
    </Col>
  </Row>
</Fragment>
`;

exports[`<MovieDetailsComponent /> should match snapshots should match snapshot - without movieDetails 1`] = `
<Fragment>
  <Row
    gutter={0}
  >
    <Col
      className="top-margin"
      offset={2}
      span={10}
    >
      <Title
        level={3}
      >
        some title
      </Title>
    </Col>
  </Row>
  <Row
    gutter={8}
    type="flex"
  >
    <Col
      className="movie__details"
      offset={2}
      span={20}
    />
  </Row>
</Fragment>
`;
